carbon-design-system/carbon-addons-iot-react
# feat(SuiteHeader): add support for on click callback for application name

Closes #3794

**Summary**

- Add option to intercept `a href="..."` with callback
- **NOTE**: disabling redirect by `evt.preventDefault()` should happen in passed callback

**Change List (commits, features, bugs, etc)**

- Add callback on application name link
- Add unit test
- Update snapshots, docs, storybook

**Acceptance Test (how to verify the PR)**

- Go to [this story](https://deploy-preview-3795--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-ui-shell-suiteheader-multi-workspace--header-with-application-link-interceptor)
- Open "Actions" tab
- Click on application name
- Verify that redirect didn't happen and callback was invoked with event and target route

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
